extensions/wikia/UserRenameTool/UserRenameTool.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * A Special Page extension to rename users, runnable by staff members.
+ * Based on the RenameUser MediaWiki extension by Arnfjörð Bjarmason and Aaron Schulz.
+ *
+ * REQUIRES THE FOLLOWING EXTENSIONS TO BE ENABLED:
+ * StaffLog, TaskManager, Phalanx, SpecialPhalanx
+ */
+
+// check for dependencies
+if ( !defined( 'MEDIAWIKI' ) ) die( "This is MediaWiki extension and cannot be used standalone." );
+
+$wgExtensionCredits['specialpage'][] = array(
+	'path' => __FILE__,
+	'name' => 'UserRenameTool',
+	'author' => [ 'Federico "Lox" Lucignano', 'Władysław Bodzek' ],
+	'url' => 'https://github.com/Wikia/app/tree/dev/extensions/wikia/UserRenameTool',
+	'description' => "Renames a user (need ''renameuser'' right)",
+	'descriptionmsg' => 'userrenametool-desc'
+);
+
+$dir = dirname( __FILE__ ) . '/';
+
+// special pages
+$wgSpecialPages['UserRenameTool'] = 'UserRenameToolController';
+$wgSpecialPageGroups['UserRenameTool'] = 'users';
+
+// internationalization files
+$wgExtensionMessagesFiles['UserRenameTool'] = $dir . 'UserRenameTool.i18n.php';
+
+// classes
+$wgAutoloadClasses['UserRenameToolController'] = $dir . 'UserRenameToolController.class.php';
+$wgAutoloadClasses['UserRenameToolHelper'] = $dir . 'UserRenameToolHelper.class.php';
+$wgAutoloadClasses['UserRenameTool\Process\ProcessBase'] = $dir . 'process/ProcessBase.class.php';
+$wgAutoloadClasses['UserRenameTool\Process\ProcessGlobal'] = $dir . 'process/ProcessGlobal.class.php';
+$wgAutoloadClasses['UserRenameTool\Process\ProcessLocal'] = $dir . 'process/ProcessLocal.class.php';
+$wgAutoloadClasses['UserRenameTool\Tasks\WikiRenameBase'] = $dir . 'tasks/WikiRenameBase.class.php';
+$wgAutoloadClasses['UserRenameTool\Tasks\MultiWikiRename'] = $dir . 'tasks/MultiWikiRename.class.php';
+$wgAutoloadClasses['UserRenameTool\Tasks\SingleWikiRename'] = $dir . 'tasks/SingleWikiRename.class.php';
+
+// log type
+global $wgLogTypes, $wgLogNames, $wgLogHeaders;
+$wgLogTypes[] = 'renameuser';
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "renameuser" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)